2025-01-06T15:25:51.1021154Z ##[group]Run pytest tardis --tardis-regression-data=/home/runner/work/tardis/tardis/tardis-regression-data --cov=tardis --cov-report=xml --cov-report=html --cov-append -m "not continuum"
2025-01-06T15:25:51.1022462Z [36;1mpytest tardis --tardis-regression-data=/home/runner/work/tardis/tardis/tardis-regression-data --cov=tardis --cov-report=xml --cov-report=html --cov-append -m "not continuum"[0m
2025-01-06T15:25:51.1049164Z shell: /usr/bin/bash -l {0}
2025-01-06T15:25:51.1049400Z env:
2025-01-06T15:25:51.1049577Z   CACHE_NUMBER: 0
2025-01-06T15:25:51.1050217Z   PYTEST_FLAGS: --tardis-regression-data=/home/runner/work/tardis/tardis/tardis-regression-data --cov=tardis --cov-report=xml --cov-report=html --cov-append
2025-01-06T15:25:51.1050878Z   CODECOV_TOKEN: 
2025-01-06T15:25:51.1051091Z   MAMBA_ROOT_PREFIX: /home/runner/micromamba
2025-01-06T15:25:51.1051386Z   MAMBA_EXE: /home/runner/micromamba-bin/micromamba
2025-01-06T15:25:51.1051732Z   CONDARC: /home/runner/work/_temp/setup-micromamba/.condarc
2025-01-06T15:25:51.1052069Z ##[endgroup]
2025-01-06T15:26:02.0314366Z ============================= test session starts ==============================
2025-01-06T15:26:02.0315323Z platform linux -- Python 3.12.4, pytest-8.2.2, pluggy-1.5.0
2025-01-06T15:26:02.0315836Z rootdir: /home/runner/work/tardis/tardis
2025-01-06T15:26:02.0316236Z configfile: pyproject.toml
2025-01-06T15:26:02.0316741Z plugins: anyio-4.4.0, doctestplus-1.2.1, cov-5.0.0, html-4.1.1, metadata-3.1.1
2025-01-06T15:26:02.0317347Z collected 1796 items / 2 deselected / 1794 selected
2025-01-06T15:26:02.0317655Z 
2025-01-06T15:26:06.8962000Z tardis/energy_input/tests/test_energy_source.py ..xxx                    [  0%]
2025-01-06T15:26:10.3405618Z tardis/energy_input/tests/test_gamma_ray_channel.py ...........          [  0%]
2025-01-06T15:26:11.9173936Z tardis/energy_input/tests/test_gamma_ray_grid.py xx.                     [  1%]
2025-01-06T15:26:13.8682310Z tardis/energy_input/tests/test_gamma_ray_interactions.py xx....          [  1%]
2025-01-06T15:26:13.8772307Z tardis/energy_input/tests/test_gamma_ray_packet_source.py sssssssss      [  1%]
2025-01-06T15:26:15.1240090Z tardis/energy_input/tests/test_util.py .....xxxx.....                    [  2%]
2025-01-06T15:26:15.4588144Z tardis/grid/tests/test_grid.py .                                         [  2%]
2025-01-06T15:26:15.4603642Z tardis/gui/tests/test_gui.py s                                           [  2%]
2025-01-06T15:26:18.8464716Z tardis/io/configuration/tests/test_config_reader.py ...........          [  3%]
2025-01-06T15:26:18.8592610Z tardis/io/configuration/tests/test_configuration_namespace.py .........  [  3%]
2025-01-06T15:26:18.8871934Z tardis/io/logger/tests/test_logging.py ssssssssssssssssssssssssssss      [  5%]
2025-01-06T15:26:55.6572174Z tardis/io/model/readers/tests/test_arepo_parser.py ..                    [  5%]
2025-01-06T15:26:55.6716714Z tardis/io/model/readers/tests/test_ascii_readers.py ...                  [  5%]
2025-01-06T15:26:55.6822643Z tardis/io/model/readers/tests/test_cmfgen_reader.py ..                   [  5%]
2025-01-06T15:26:55.7117680Z tardis/io/model/readers/tests/test_csvy_reader.py ....                   [  6%]
2025-01-06T15:26:55.7338136Z tardis/io/model/readers/tests/test_stella_reader.py ..                   [  6%]
2025-01-06T15:26:56.0303954Z tardis/io/tests/test_HDFWriter.py ...............                        [  7%]
2025-01-06T15:26:56.3439055Z tardis/io/tests/test_atomic.py .....                                     [  7%]
2025-01-06T15:26:56.3629463Z tardis/io/tests/test_decay.py ..                                         [  7%]
2025-01-06T15:26:56.5517020Z tardis/io/tests/test_model_reader.py ......                              [  7%]
2025-01-06T15:26:56.5584990Z tardis/model/geometry/tests/test_radial1d.py ..                          [  7%]
2025-01-06T15:27:53.0934782Z tardis/model/tests/test_base.py ........................................ [ 10%]
2025-01-06T15:27:54.0654743Z ............                                                             [ 10%]
2025-01-06T15:27:57.0697070Z tardis/model/tests/test_csvy_model.py .........                          [ 11%]
2025-01-06T15:27:57.1174849Z tardis/model/tests/test_density.py ..                                    [ 11%]
2025-01-06T15:27:58.3972468Z tardis/opacities/tests/test_opacity_solver.py .....                      [ 11%]
2025-01-06T15:27:59.7095081Z tardis/opacities/tests/test_opacity_state_numba.py .....                 [ 11%]
2025-01-06T15:28:01.6742300Z tardis/opacities/tests/test_tau_sobolev.py ..                            [ 12%]
2025-01-06T15:28:04.3625482Z tardis/plasma/equilibrium/tests/test_collisional_transitions.py ...      [ 12%]
2025-01-06T15:28:50.7961015Z tardis/plasma/equilibrium/tests/test_level_populations.py ....           [ 12%]
2025-01-06T15:29:02.5469131Z tardis/plasma/equilibrium/tests/test_rate_matrix.py .                    [ 12%]
2025-01-06T15:29:03.9604362Z tardis/plasma/tests/test_complete_plasmas.py .....................s..... [ 13%]
2025-01-06T15:29:06.8569263Z s....ssssss............s............ss.....s....ssssss............s..... [ 18%]
2025-01-06T15:29:08.5696967Z .......ss.....s....ssssss............s............ss.....s....ssssss.... [ 22%]
2025-01-06T15:29:11.4756232Z ........s............ss.....s........ss............s............ss.....s [ 26%]
2025-01-06T15:29:14.3794967Z ........ss............s............ss.....s....ssssss............s...... [ 30%]
2025-01-06T15:29:15.1305442Z ......ss.....s....ssssss............s............ss.....s.ss.ssssss..... [ 34%]
2025-01-06T15:29:18.0117985Z .......s............ss.....s....ssssss............s............ss.....s. [ 38%]
2025-01-06T15:29:21.2475461Z ...ssssss............s............ss.....s....ssssss............s....... [ 42%]
2025-01-06T15:29:24.3245579Z .....ss.....s....ssssss............s............s...........ssssss...... [ 46%]
2025-01-06T15:29:26.1542880Z ......s............s...........ssssss............s............s......... [ 50%]
2025-01-06T15:29:29.0744106Z ..ssssss............s............ss.....s....ssssss............s........ [ 54%]
2025-01-06T15:29:32.2535953Z ....ss.....s....ssssss..........................s.....s....ssss......... [ 58%]
2025-01-06T15:29:32.5838833Z ...................s.....s....ssss.......                                [ 60%]
2025-01-06T15:29:32.6988873Z tardis/plasma/tests/test_hdf_plasma.py ................................. [ 62%]
2025-01-06T15:29:32.7108731Z ......                                                                   [ 62%]
2025-01-06T15:29:32.8474876Z tardis/plasma/tests/test_nlte_excitation.py ...                          [ 62%]
2025-01-06T15:29:49.7042292Z tardis/plasma/tests/test_nlte_solver.py ..EEEx                           [ 63%]
2025-01-06T15:29:49.7067119Z tardis/plasma/tests/test_plasma_continuum.py .                           [ 63%]
2025-01-06T15:30:12.5761844Z tardis/plasma/tests/test_plasma_vboundary.py .............               [ 63%]
2025-01-06T15:30:26.3435317Z tardis/plasma/tests/test_tardis_model_density_config.py ...              [ 64%]
2025-01-06T15:30:27.0828652Z tardis/simulation/tests/test_convergence.py .....                        [ 64%]
2025-01-06T15:30:35.7756012Z tardis/simulation/tests/test_simulation.py .............                 [ 65%]
2025-01-06T15:30:35.7969675Z tardis/spectrum/tests/test_cuda_formal_integral.py sssssssssssssssssssss [ 66%]
2025-01-06T15:30:36.2642789Z ssss...ssssssssssssssssssssss                                            [ 67%]
2025-01-06T15:30:36.2716548Z tardis/spectrum/tests/test_luminosity.py ....                            [ 68%]
2025-01-06T15:30:37.8728027Z tardis/spectrum/tests/test_numba_formal_integral.py .................... [ 69%]
2025-01-06T15:30:37.8921089Z ..........                                                               [ 69%]
2025-01-06T15:30:38.0630899Z tardis/spectrum/tests/test_spectrum.py ................................. [ 71%]
2025-01-06T15:30:38.0959556Z .........                                                                [ 72%]
2025-01-06T15:30:57.4379346Z tardis/spectrum/tests/test_spectrum_solver.py ....                       [ 72%]
2025-01-06T15:31:58.8083786Z tardis/tests/test_tardis_full.py ....                                    [ 72%]
2025-01-06T15:33:22.7923016Z tardis/tests/test_tardis_full_formal_integral.py ................        [ 73%]
2025-01-06T15:33:22.8688789Z tardis/tests/test_util.py .............................................  [ 75%]
2025-01-06T15:33:25.5037128Z tardis/transport/montecarlo/tests/test_base.py x.....................    [ 77%]
2025-01-06T15:33:33.1499127Z tardis/transport/montecarlo/tests/test_interaction.py .......            [ 77%]
2025-01-06T15:33:33.3619009Z tardis/transport/montecarlo/tests/test_macro_atom.py ....                [ 77%]
2025-01-06T15:33:33.3794996Z tardis/transport/montecarlo/tests/test_montecarlo.py sssssssssssssssssss [ 78%]
2025-01-06T15:33:33.4211177Z sssssssssssssssssssssssssssssssssssssssssssss                            [ 81%]
2025-01-06T15:33:42.2617755Z tardis/transport/montecarlo/tests/test_montecarlo_main_loop.py x..       [ 81%]
2025-01-06T15:33:42.2685043Z tardis/transport/montecarlo/tests/test_nonhomologous.py ...              [ 81%]
2025-01-06T15:33:47.9187400Z tardis/transport/montecarlo/tests/test_numba_interface.py .....x.        [ 82%]
2025-01-06T15:33:48.3971603Z tardis/transport/montecarlo/tests/test_opacities.py ...............      [ 82%]
2025-01-06T15:33:49.4051654Z tardis/transport/montecarlo/tests/test_packet.py .................xXXXXX [ 84%]
2025-01-06T15:33:49.5357328Z XX.........                                                              [ 84%]
2025-01-06T15:33:49.5637029Z tardis/transport/montecarlo/tests/test_packet_source.py ........         [ 85%]
2025-01-06T15:33:49.8647468Z tardis/transport/montecarlo/tests/test_rpacket_last_interaction_tracker.py . [ 85%]
2025-01-06T15:33:50.7411194Z .....                                                                    [ 85%]
2025-01-06T15:34:24.8817722Z tardis/transport/montecarlo/tests/test_rpacket_tracker.py ......         [ 85%]
2025-01-06T15:34:24.9103227Z tardis/transport/montecarlo/tests/test_single_packet_loop.py xxx         [ 86%]
2025-01-06T15:34:24.9135104Z tardis/transport/montecarlo/tests/test_tracker_utils.py ..               [ 86%]
2025-01-06T15:34:27.7506110Z tardis/transport/montecarlo/tests/test_vpacket.py ..x.                   [ 86%]
2025-01-06T15:34:27.7678202Z tardis/transport/montecarlo/tests/test_weighted_packet_source.py ....    [ 86%]
2025-01-06T15:34:31.2019436Z tardis/transport/montecarlo/tests/test_weighted_packet_source_integration.py . [ 86%]
2025-01-06T15:34:31.2021579Z                                                                          [ 86%]
2025-01-06T15:34:31.9544757Z tardis/transport/tests/test_doppler_factor.py ..................         [ 87%]
2025-01-06T15:34:33.3032212Z tardis/visualization/tools/tests/test_convergence_plot.py .............. [ 88%]
2025-01-06T15:34:35.4921522Z ......                                                                   [ 88%]
2025-01-06T15:34:40.2272580Z tardis/visualization/tools/tests/test_liv_plot.py ...................... [ 90%]
2025-01-06T15:34:43.3529519Z .............................                                            [ 91%]
2025-01-06T15:34:46.1908831Z tardis/visualization/tools/tests/test_rpacket_plot.py .................. [ 92%]
2025-01-06T15:34:53.2013647Z ....                                                                     [ 92%]
2025-01-06T15:34:59.1801486Z tardis/visualization/tools/tests/test_sdec_plot.py ..................... [ 94%]
2025-01-06T15:35:06.7495122Z ..............................                                           [ 95%]
2025-01-06T15:35:07.1810599Z tardis/visualization/widgets/tests/test_custom_abundance.py ........ss.. [ 96%]
2025-01-06T15:35:07.2521734Z ....                                                                     [ 96%]
2025-01-06T15:35:08.2981951Z tardis/visualization/widgets/tests/test_line_info.py ................... [ 97%]
2025-01-06T15:35:08.9218993Z ..................                                                       [ 98%]
2025-01-06T15:36:12.4790539Z tardis/visualization/widgets/tests/test_shell_info.py .................. [ 99%]
2025-01-06T15:36:20.6261761Z ......                                                                   [100%]
2025-01-06T15:36:20.6262242Z 
2025-01-06T15:36:20.6262452Z ==================================== ERRORS ====================================
2025-01-06T15:36:20.6263141Z _________ ERROR at setup of test_critical_case_dilution_factor_1_root __________
2025-01-06T15:36:20.6263632Z 
2025-01-06T15:36:20.6265205Z tardis_model_config_nlte_root = {'tardis_config_version': 'v1.0', 'supernova': {'luminosity_requested': <Quantity 4.31528298e+42 erg / s>, 'time_explo....0, 'enable_biasing': False, 'virtual_packet_logging': False}}, 'config_dirname': 'tardis/io/configuration/tests/data'}
2025-01-06T15:36:20.6267160Z nlte_raw_model_root = <tardis.model.base.SimulationState object at 0x7f0231a2b6e0>
2025-01-06T15:36:20.6268045Z nlte_atom_data = <Atomic Data UUID=None MD5=None Lines=69056 Levels=26012>
2025-01-06T15:36:20.6268545Z 
2025-01-06T15:36:20.6268701Z     @pytest.fixture
2025-01-06T15:36:20.6269112Z     def nlte_raw_plasma_dilution_factor_1_root(
2025-01-06T15:36:20.6269839Z         tardis_model_config_nlte_root, nlte_raw_model_root, nlte_atom_data
2025-01-06T15:36:20.6270439Z     ):
2025-01-06T15:36:20.6270725Z         """
2025-01-06T15:36:20.6271116Z         Plasma assembled with dilution factors set to 1.0.
2025-01-06T15:36:20.6271631Z         """
2025-01-06T15:36:20.6272138Z         new_dilution_factor = np.ones_like(nlte_raw_model_root.dilution_factor)
2025-01-06T15:36:20.6272905Z         nlte_raw_model_root.dilution_factor = new_dilution_factor
2025-01-06T15:36:20.6273486Z >       plasma = assemble_plasma(
2025-01-06T15:36:20.6274361Z             tardis_model_config_nlte_root, nlte_raw_model_root, nlte_atom_data
2025-01-06T15:36:20.6275218Z         )
2025-01-06T15:36:20.6275396Z 
2025-01-06T15:36:20.6275610Z tardis/plasma/tests/test_nlte_solver.py:232: 
2025-01-06T15:36:20.6276163Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-01-06T15:36:20.6276865Z tardis/plasma/assembly/legacy_assembly.py:37: in assemble_plasma
2025-01-06T15:36:20.6277539Z     return plasma_solver_factory.assemble(
2025-01-06T15:36:20.6278086Z tardis/plasma/assembly/base.py:614: in assemble
2025-01-06T15:36:20.6278605Z     return BasePlasma(
2025-01-06T15:36:20.6279006Z tardis/io/util.py:195: in __new__
2025-01-06T15:36:20.6279460Z     instance.__init__(*args, **kwargs)
2025-01-06T15:36:20.6279939Z tardis/plasma/base.py:40: in __init__
2025-01-06T15:36:20.6280415Z     self.update(**kwargs)
2025-01-06T15:36:20.6280835Z tardis/plasma/base.py:192: in update
2025-01-06T15:36:20.6281374Z     self.plasma_properties_dict[module_name].update()
2025-01-06T15:36:20.6281979Z tardis/plasma/properties/base.py:110: in update
2025-01-06T15:36:20.6282538Z     self.calculate(*self._get_input_values()),
2025-01-06T15:36:20.6283256Z tardis/plasma/properties/continuum_processes/rates.py:231: in calculate
2025-01-06T15:36:20.6284021Z     alpha_sp_E = cumulative_integrate_array_by_blocks(
2025-01-06T15:36:20.6285158Z ../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/dispatcher.py:468: in _compile_for_args
2025-01-06T15:36:20.6286025Z     error_rewrite(e, 'typing')
2025-01-06T15:36:20.6286472Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-01-06T15:36:20.6286883Z 
2025-01-06T15:36:20.6288042Z e = TypingError('Failed in nopython mode pipeline (step: nopython frontend)\n\x1b[1m\x1b[1m\x1b[1m\x1b[1mFailed in nopytho...[1m            integrated[start:stop, col] = numba_cumulative_trapezoid(\n\x1b[0m            \x1b[1m^\x1b[0m\x1b[0m\n')
2025-01-06T15:36:20.6289413Z issue_type = 'typing'
2025-01-06T15:36:20.6289633Z 
2025-01-06T15:36:20.6289785Z     def error_rewrite(e, issue_type):
2025-01-06T15:36:20.6290175Z         """
2025-01-06T15:36:20.6290597Z         Rewrite and raise Exception `e` with help supplied based on the
2025-01-06T15:36:20.6291141Z         specified issue_type.
2025-01-06T15:36:20.6291758Z         """
2025-01-06T15:36:20.6292057Z         if config.SHOW_HELP:
2025-01-06T15:36:20.6292471Z             help_msg = errors.error_extras[issue_type]
2025-01-06T15:36:20.6293020Z             e.patch_message('\n'.join((str(e).rstrip(), help_msg)))
2025-01-06T15:36:20.6293555Z         if config.FULL_TRACEBACKS:
2025-01-06T15:36:20.6293956Z             raise e
2025-01-06T15:36:20.6294260Z         else:
2025-01-06T15:36:20.6294792Z >           raise e.with_traceback(None)
2025-01-06T15:36:20.6295508Z E           numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6296684Z E           [1m[1m[1m[1mFailed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6297794Z E           [1m[1m[1mNo implementation of function Function(<built-in function concatenate>) found for signature:
2025-01-06T15:36:20.6298541Z E            
2025-01-06T15:36:20.6299006Z E            >>> concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))
2025-01-06T15:36:20.6299609Z E            
2025-01-06T15:36:20.6299977Z E           There are 2 candidate implementations:
2025-01-06T15:36:20.6300606Z E           [1m  - Of which 1 did not match due to:
2025-01-06T15:36:20.6301295Z E             Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:36:20.6302283Z E               With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)))':[0m
2025-01-06T15:36:20.6303150Z E           [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6303948Z E                TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6305511Z E              [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:36:20.6306172Z E               
2025-01-06T15:36:20.6306656Z E               >>> np_concatenate(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:36:20.6307238Z E               
2025-01-06T15:36:20.6307555Z E              There are 2 candidate implementations:
2025-01-06T15:36:20.6308114Z E              [1m      - Of which 2 did not match due to:
2025-01-06T15:36:20.6308791Z E                    Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:36:20.6309763Z E                      With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:36:20.6310585Z E              [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6311728Z E                       TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:36:20.6313014Z E                raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:36:20.6313897Z E              [0m
2025-01-06T15:36:20.6314739Z E              [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:36:20.6316007Z E              [0m[1mDuring: typing of call at /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:36:20.6316915Z E              [0m
2025-01-06T15:36:20.6317253Z E              [1m
2025-01-06T15:36:20.6318050Z E              File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:36:20.6318898Z E              [1m        def impl(arrays, axis=0):
2025-01-06T15:36:20.6319480Z E              [1m            return np_concatenate(arrays, axis)
2025-01-06T15:36:20.6320017Z E              [0m            [1m^[0m[0m
2025-01-06T15:36:20.6320444Z E           [0m
2025-01-06T15:36:20.6321170Z E             raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:36:20.6322095Z E           [1m  - Of which 1 did not match due to:
2025-01-06T15:36:20.6322975Z E             Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:36:20.6323868Z E               With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))':[0m
2025-01-06T15:36:20.6324844Z E           [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6325609Z E                TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6326624Z E              [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:36:20.6327281Z E               
2025-01-06T15:36:20.6327772Z E               >>> np_concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:36:20.6328336Z E               
2025-01-06T15:36:20.6328653Z E              There are 2 candidate implementations:
2025-01-06T15:36:20.6329180Z E              [1m      - Of which 1 did not match due to:
2025-01-06T15:36:20.6329804Z E                    Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:36:20.6330685Z E                      With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:36:20.6331462Z E              [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6332552Z E                       TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:36:20.6333756Z E                raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:36:20.6335061Z E              [1m      - Of which 1 did not match due to:
2025-01-06T15:36:20.6335681Z E                    Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:36:20.6336596Z E                      With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))':[0m
2025-01-06T15:36:20.6337392Z E              [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6338428Z E                       TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=[0]>, array(float64, 1d, C))[0m
2025-01-06T15:36:20.6339619Z E                raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:36:20.6340419Z E              [0m
2025-01-06T15:36:20.6341000Z E              [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:36:20.6342195Z E              [0m[1mDuring: typing of call at /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:36:20.6343090Z E              [0m
2025-01-06T15:36:20.6343401Z E              [1m
2025-01-06T15:36:20.6344185Z E              File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:36:20.6345187Z E              [1m        def impl(arrays, axis=0):
2025-01-06T15:36:20.6345734Z E              [1m            return np_concatenate(arrays, axis)
2025-01-06T15:36:20.6346228Z E              [0m            [1m^[0m[0m
2025-01-06T15:36:20.6346628Z E           [0m
2025-01-06T15:36:20.6347288Z E             raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:36:20.6348069Z E           [0m
2025-01-06T15:36:20.6348646Z E           [0m[1mDuring: resolving callee type: Function(<built-in function concatenate>)[0m
2025-01-06T15:36:20.6349923Z E           [0m[1mDuring: typing of call at /home/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (36)
2025-01-06T15:36:20.6350846Z E           [0m
2025-01-06T15:36:20.6351138Z E           [1m
2025-01-06T15:36:20.6351758Z E           File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 36:[0m
2025-01-06T15:36:20.6352689Z E           [1mdef numba_cumulative_trapezoid(f, x):
2025-01-06T15:36:20.6353097Z E               <source elided>
2025-01-06T15:36:20.6353432Z E           
2025-01-06T15:36:20.6353940Z E           [1m    return np.concatenate(([0], cumulative / cumulative[-1]))
2025-01-06T15:36:20.6354642Z E           [0m    [1m^[0m[0m
2025-01-06T15:36:20.6354978Z E           
2025-01-06T15:36:20.6355789Z E           [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x7f023c3bd080>))[0m
2025-01-06T15:36:20.6357243Z E           [0m[1mDuring: typing of call at /home/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:36:20.6358215Z E           [0m
2025-01-06T15:36:20.6359028Z E           [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x7f023c3bd080>))[0m
2025-01-06T15:36:20.6360493Z E           [0m[1mDuring: typing of call at /home/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:36:20.6361423Z E           [0m
2025-01-06T15:36:20.6361721Z E           [1m
2025-01-06T15:36:20.6362366Z E           File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 68:[0m
2025-01-06T15:36:20.6363260Z E           [1mdef cumulative_integrate_array_by_blocks(f, x, block_references):
2025-01-06T15:36:20.6363786Z E               <source elided>
2025-01-06T15:36:20.6364109Z E           
2025-01-06T15:36:20.6364739Z E           [1m            integrated[start:stop, col] = numba_cumulative_trapezoid(
2025-01-06T15:36:20.6365501Z E           [0m            [1m^[0m[0m
2025-01-06T15:36:20.6365760Z 
2025-01-06T15:36:20.6366204Z ../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/dispatcher.py:409: TypingError
2025-01-06T15:36:20.6367050Z ---------------------------- Captured stdout setup -----------------------------
2025-01-06T15:36:20.6367812Z [[1mtardis.io.model.parse_density_configuration[0m][[1;33mWARNING[0m]  
2025-01-06T15:36:20.6368937Z 	Number of density points larger than number of shells. Assuming inner point irrelevant ([1mparse_density_configuration.py[0m:114)
2025-01-06T15:36:20.6369847Z ------------------------------ Captured log setup ------------------------------
2025-01-06T15:36:20.6370986Z WARNING  tardis.io.model.parse_density_configuration:parse_density_configuration.py:114 Number of density points larger than number of shells. Assuming inner point irrelevant
2025-01-06T15:36:20.6372219Z __________ ERROR at setup of test_critical_case_dilution_factor_1_lu ___________
2025-01-06T15:36:20.6372632Z 
2025-01-06T15:36:20.6373749Z tardis_model_config_nlte_lu = {'tardis_config_version': 'v1.0', 'supernova': {'luminosity_requested': <Quantity 4.31528298e+42 erg / s>, 'time_explo....0, 'enable_biasing': False, 'virtual_packet_logging': False}}, 'config_dirname': 'tardis/io/configuration/tests/data'}
2025-01-06T15:36:20.6375447Z nlte_raw_model_lu = <tardis.model.base.SimulationState object at 0x7f02307ede20>
2025-01-06T15:36:20.6376153Z nlte_atom_data = <Atomic Data UUID=None MD5=None Lines=69056 Levels=26012>
2025-01-06T15:36:20.6376566Z 
2025-01-06T15:36:20.6376684Z     @pytest.fixture
2025-01-06T15:36:20.6377001Z     def nlte_raw_plasma_dilution_factor_1_lu(
2025-01-06T15:36:20.6377502Z         tardis_model_config_nlte_lu, nlte_raw_model_lu, nlte_atom_data
2025-01-06T15:36:20.6377972Z     ):
2025-01-06T15:36:20.6378215Z         """
2025-01-06T15:36:20.6378569Z         Plasma assembled with dilution factors set to 1.0.
2025-01-06T15:36:20.6379001Z         """
2025-01-06T15:36:20.6379424Z         new_dilution_factor = np.ones_like(nlte_raw_model_lu.dilution_factor)
2025-01-06T15:36:20.6380064Z         nlte_raw_model_lu.dilution_factor = new_dilution_factor
2025-01-06T15:36:20.6380533Z >       plasma = assemble_plasma(
2025-01-06T15:36:20.6381009Z             tardis_model_config_nlte_lu, nlte_raw_model_lu, nlte_atom_data
2025-01-06T15:36:20.6381702Z         )
2025-01-06T15:36:20.6381842Z 
2025-01-06T15:36:20.6382012Z tardis/plasma/tests/test_nlte_solver.py:247: 
2025-01-06T15:36:20.6382465Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-01-06T15:36:20.6383054Z tardis/plasma/assembly/legacy_assembly.py:37: in assemble_plasma
2025-01-06T15:36:20.6383613Z     return plasma_solver_factory.assemble(
2025-01-06T15:36:20.6384065Z tardis/plasma/assembly/base.py:614: in assemble
2025-01-06T15:36:20.6384645Z     return BasePlasma(
2025-01-06T15:36:20.6384972Z tardis/io/util.py:195: in __new__
2025-01-06T15:36:20.6385328Z     instance.__init__(*args, **kwargs)
2025-01-06T15:36:20.6385726Z tardis/plasma/base.py:40: in __init__
2025-01-06T15:36:20.6386104Z     self.update(**kwargs)
2025-01-06T15:36:20.6386436Z tardis/plasma/base.py:192: in update
2025-01-06T15:36:20.6386919Z     self.plasma_properties_dict[module_name].update()
2025-01-06T15:36:20.6387397Z tardis/plasma/properties/base.py:110: in update
2025-01-06T15:36:20.6387853Z     self.calculate(*self._get_input_values()),
2025-01-06T15:36:20.6388447Z tardis/plasma/properties/continuum_processes/rates.py:231: in calculate
2025-01-06T15:36:20.6389077Z     alpha_sp_E = cumulative_integrate_array_by_blocks(
2025-01-06T15:36:20.6389913Z ../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/dispatcher.py:468: in _compile_for_args
2025-01-06T15:36:20.6390697Z     error_rewrite(e, 'typing')
2025-01-06T15:36:20.6391081Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-01-06T15:36:20.6391386Z 
2025-01-06T15:36:20.6392310Z e = TypingError('Failed in nopython mode pipeline (step: nopython frontend)\n\x1b[1m\x1b[1m\x1b[1m\x1b[1mFailed in nopytho...[1m            integrated[start:stop, col] = numba_cumulative_trapezoid(\n\x1b[0m            \x1b[1m^\x1b[0m\x1b[0m\n')
2025-01-06T15:36:20.6393655Z issue_type = 'typing'
2025-01-06T15:36:20.6393833Z 
2025-01-06T15:36:20.6393977Z     def error_rewrite(e, issue_type):
2025-01-06T15:36:20.6394326Z         """
2025-01-06T15:36:20.6394865Z         Rewrite and raise Exception `e` with help supplied based on the
2025-01-06T15:36:20.6395369Z         specified issue_type.
2025-01-06T15:36:20.6395686Z         """
2025-01-06T15:36:20.6395943Z         if config.SHOW_HELP:
2025-01-06T15:36:20.6396311Z             help_msg = errors.error_extras[issue_type]
2025-01-06T15:36:20.6396800Z             e.patch_message('\n'.join((str(e).rstrip(), help_msg)))
2025-01-06T15:36:20.6397265Z         if config.FULL_TRACEBACKS:
2025-01-06T15:36:20.6397616Z             raise e
2025-01-06T15:36:20.6397879Z         else:
2025-01-06T15:36:20.6398175Z >           raise e.with_traceback(None)
2025-01-06T15:36:20.6398882Z E           numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6399802Z E           [1m[1m[1m[1mFailed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6400746Z E           [1m[1m[1mNo implementation of function Function(<built-in function concatenate>) found for signature:
2025-01-06T15:36:20.6401412Z E            
2025-01-06T15:36:20.6401794Z E            >>> concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))
2025-01-06T15:36:20.6402298Z E            
2025-01-06T15:36:20.6402607Z E           There are 2 candidate implementations:
2025-01-06T15:36:20.6403129Z E           [1m  - Of which 1 did not match due to:
2025-01-06T15:36:20.6403759Z E             Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:36:20.6404905Z E               With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)))':[0m
2025-01-06T15:36:20.6405648Z E           [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6406344Z E                TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6407301Z E              [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:36:20.6408178Z E               
2025-01-06T15:36:20.6408757Z E               >>> np_concatenate(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:36:20.6409410Z E               
2025-01-06T15:36:20.6409727Z E              There are 2 candidate implementations:
2025-01-06T15:36:20.6410262Z E              [1m      - Of which 2 did not match due to:
2025-01-06T15:36:20.6410878Z E                    Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:36:20.6411783Z E                      With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:36:20.6412570Z E              [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6413641Z E                       TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:36:20.6415036Z E                raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:36:20.6415888Z E              [0m
2025-01-06T15:36:20.6416463Z E              [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:36:20.6417614Z E              [0m[1mDuring: typing of call at /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:36:20.6418465Z E              [0m
2025-01-06T15:36:20.6418761Z E              [1m
2025-01-06T15:36:20.6419491Z E              File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:36:20.6420578Z E              [1m        def impl(arrays, axis=0):
2025-01-06T15:36:20.6421095Z E              [1m            return np_concatenate(arrays, axis)
2025-01-06T15:36:20.6421579Z E              [0m            [1m^[0m[0m
2025-01-06T15:36:20.6421973Z E           [0m
2025-01-06T15:36:20.6422646Z E             raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:36:20.6423486Z E           [1m  - Of which 1 did not match due to:
2025-01-06T15:36:20.6424090Z E             Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:36:20.6425100Z E               With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))':[0m
2025-01-06T15:36:20.6425834Z E           [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6426494Z E                TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6427439Z E              [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:36:20.6428066Z E               
2025-01-06T15:36:20.6428518Z E               >>> np_concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:36:20.6429079Z E               
2025-01-06T15:36:20.6429391Z E              There are 2 candidate implementations:
2025-01-06T15:36:20.6429925Z E              [1m      - Of which 1 did not match due to:
2025-01-06T15:36:20.6430530Z E                    Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:36:20.6431598Z E                      With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:36:20.6432362Z E              [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6433421Z E                       TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:36:20.6434765Z E                raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:36:20.6435684Z E              [1m      - Of which 1 did not match due to:
2025-01-06T15:36:20.6436472Z E                    Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:36:20.6437387Z E                      With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))':[0m
2025-01-06T15:36:20.6438183Z E              [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6439234Z E                       TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=[0]>, array(float64, 1d, C))[0m
2025-01-06T15:36:20.6440420Z E                raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:36:20.6441249Z E              [0m
2025-01-06T15:36:20.6441819Z E              [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:36:20.6442979Z E              [0m[1mDuring: typing of call at /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:36:20.6443888Z E              [0m
2025-01-06T15:36:20.6444210Z E              [1m
2025-01-06T15:36:20.6445122Z E              File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:36:20.6445682Z E              [1m        def impl(arrays, axis=0):
2025-01-06T15:36:20.6446053Z E              [1m            return np_concatenate(arrays, axis)
2025-01-06T15:36:20.6446386Z E              [0m            [1m^[0m[0m
2025-01-06T15:36:20.6446653Z E           [0m
2025-01-06T15:36:20.6447086Z E             raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:36:20.6447741Z E           [0m
2025-01-06T15:36:20.6448137Z E           [0m[1mDuring: resolving callee type: Function(<built-in function concatenate>)[0m
2025-01-06T15:36:20.6448929Z E           [0m[1mDuring: typing of call at /home/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (36)
2025-01-06T15:36:20.6449508Z E           [0m
2025-01-06T15:36:20.6449708Z E           [1m
2025-01-06T15:36:20.6450124Z E           File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 36:[0m
2025-01-06T15:36:20.6450604Z E           [1mdef numba_cumulative_trapezoid(f, x):
2025-01-06T15:36:20.6450875Z E               <source elided>
2025-01-06T15:36:20.6451089Z E           
2025-01-06T15:36:20.6451402Z E           [1m    return np.concatenate(([0], cumulative / cumulative[-1]))
2025-01-06T15:36:20.6451758Z E           [0m    [1m^[0m[0m
2025-01-06T15:36:20.6451975Z E           
2025-01-06T15:36:20.6452484Z E           [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x7f023c3bd080>))[0m
2025-01-06T15:36:20.6453376Z E           [0m[1mDuring: typing of call at /home/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:36:20.6453952Z E           [0m
2025-01-06T15:36:20.6454659Z E           [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x7f023c3bd080>))[0m
2025-01-06T15:36:20.6455571Z E           [0m[1mDuring: typing of call at /home/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:36:20.6456154Z E           [0m
2025-01-06T15:36:20.6456362Z E           [1m
2025-01-06T15:36:20.6456781Z E           File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 68:[0m
2025-01-06T15:36:20.6457345Z E           [1mdef cumulative_integrate_array_by_blocks(f, x, block_references):
2025-01-06T15:36:20.6457701Z E               <source elided>
2025-01-06T15:36:20.6457910Z E           
2025-01-06T15:36:20.6458227Z E           [1m            integrated[start:stop, col] = numba_cumulative_trapezoid(
2025-01-06T15:36:20.6458593Z E           [0m            [1m^[0m[0m
2025-01-06T15:36:20.6458887Z 
2025-01-06T15:36:20.6459178Z ../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/dispatcher.py:409: TypingError
2025-01-06T15:36:20.6459703Z ---------------------------- Captured stdout setup -----------------------------
2025-01-06T15:36:20.6460192Z [[1mtardis.io.model.parse_density_configuration[0m][[1;33mWARNING[0m]  
2025-01-06T15:36:20.6460911Z 	Number of density points larger than number of shells. Assuming inner point irrelevant ([1mparse_density_configuration.py[0m:114)
2025-01-06T15:36:20.6461544Z [[1mtardis.plasma.assembly.base[0m][[1;33mWARNING[0m]  
2025-01-06T15:36:20.6462074Z 	LU solver will be inaccurate for NLTE excitation, proceed with caution. ([1mbase.py[0m:489)
2025-01-06T15:36:20.6462552Z ------------------------------ Captured log setup ------------------------------
2025-01-06T15:36:20.6463289Z WARNING  tardis.io.model.parse_density_configuration:parse_density_configuration.py:114 Number of density points larger than number of shells. Assuming inner point irrelevant
2025-01-06T15:36:20.6464217Z WARNING  tardis.plasma.assembly.base:base.py:489 LU solver will be inaccurate for NLTE excitation, proceed with caution.
2025-01-06T15:36:20.6464986Z _________ ERROR at setup of test_critical_case_dilution_factor_0_root __________
2025-01-06T15:36:20.6465248Z 
2025-01-06T15:36:20.6465951Z tardis_model_config_nlte_root = {'tardis_config_version': 'v1.0', 'supernova': {'luminosity_requested': <Quantity 4.31528298e+42 erg / s>, 'time_explo....0, 'enable_biasing': False, 'virtual_packet_logging': False}}, 'config_dirname': 'tardis/io/configuration/tests/data'}
2025-01-06T15:36:20.6466930Z nlte_raw_model_root = <tardis.model.base.SimulationState object at 0x7f022b514830>
2025-01-06T15:36:20.6467566Z nlte_atom_data = <Atomic Data UUID=None MD5=None Lines=69056 Levels=26012>
2025-01-06T15:36:20.6467825Z 
2025-01-06T15:36:20.6467907Z     @pytest.fixture
2025-01-06T15:36:20.6468145Z     def nlte_raw_plasma_dilution_factor_0_root(
2025-01-06T15:36:20.6468510Z         tardis_model_config_nlte_root, nlte_raw_model_root, nlte_atom_data
2025-01-06T15:36:20.6468842Z     ):
2025-01-06T15:36:20.6469006Z         """
2025-01-06T15:36:20.6469230Z         Plasma assembled with dilution factors set to 0.0.
2025-01-06T15:36:20.6469508Z         """
2025-01-06T15:36:20.6469788Z         new_dilution_factor = np.zeros_like(nlte_raw_model_root.dilution_factor)
2025-01-06T15:36:20.6470203Z         nlte_raw_model_root.dilution_factor = new_dilution_factor
2025-01-06T15:36:20.6470522Z >       plasma = assemble_plasma(
2025-01-06T15:36:20.6470845Z             tardis_model_config_nlte_root, nlte_raw_model_root, nlte_atom_data
2025-01-06T15:36:20.6471170Z         )
2025-01-06T15:36:20.6471268Z 
2025-01-06T15:36:20.6471383Z tardis/plasma/tests/test_nlte_solver.py:262: 
2025-01-06T15:36:20.6471690Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-01-06T15:36:20.6472062Z tardis/plasma/assembly/legacy_assembly.py:37: in assemble_plasma
2025-01-06T15:36:20.6472431Z     return plasma_solver_factory.assemble(
2025-01-06T15:36:20.6472722Z tardis/plasma/assembly/base.py:614: in assemble
2025-01-06T15:36:20.6472995Z     return BasePlasma(
2025-01-06T15:36:20.6473211Z tardis/io/util.py:195: in __new__
2025-01-06T15:36:20.6473452Z     instance.__init__(*args, **kwargs)
2025-01-06T15:36:20.6473714Z tardis/plasma/base.py:40: in __init__
2025-01-06T15:36:20.6473962Z     self.update(**kwargs)
2025-01-06T15:36:20.6474187Z tardis/plasma/base.py:192: in update
2025-01-06T15:36:20.6474579Z     self.plasma_properties_dict[module_name].update()
2025-01-06T15:36:20.6474901Z tardis/plasma/properties/base.py:110: in update
2025-01-06T15:36:20.6475199Z     self.calculate(*self._get_input_values()),
2025-01-06T15:36:20.6475582Z tardis/plasma/properties/continuum_processes/rates.py:231: in calculate
2025-01-06T15:36:20.6475982Z     alpha_sp_E = cumulative_integrate_array_by_blocks(
2025-01-06T15:36:20.6476501Z ../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/dispatcher.py:468: in _compile_for_args
2025-01-06T15:36:20.6477106Z     error_rewrite(e, 'typing')
2025-01-06T15:36:20.6477368Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-01-06T15:36:20.6477571Z 
2025-01-06T15:36:20.6478185Z e = TypingError('Failed in nopython mode pipeline (step: nopython frontend)\n\x1b[1m\x1b[1m\x1b[1m\x1b[1mFailed in nopytho...[1m            integrated[start:stop, col] = numba_cumulative_trapezoid(\n\x1b[0m            \x1b[1m^\x1b[0m\x1b[0m\n')
2025-01-06T15:36:20.6478948Z issue_type = 'typing'
2025-01-06T15:36:20.6479071Z 
2025-01-06T15:36:20.6479166Z     def error_rewrite(e, issue_type):
2025-01-06T15:36:20.6479401Z         """
2025-01-06T15:36:20.6479664Z         Rewrite and raise Exception `e` with help supplied based on the
2025-01-06T15:36:20.6479991Z         specified issue_type.
2025-01-06T15:36:20.6480206Z         """
2025-01-06T15:36:20.6480387Z         if config.SHOW_HELP:
2025-01-06T15:36:20.6480637Z             help_msg = errors.error_extras[issue_type]
2025-01-06T15:36:20.6480972Z             e.patch_message('\n'.join((str(e).rstrip(), help_msg)))
2025-01-06T15:36:20.6481276Z         if config.FULL_TRACEBACKS:
2025-01-06T15:36:20.6481505Z             raise e
2025-01-06T15:36:20.6481686Z         else:
2025-01-06T15:36:20.6481878Z >           raise e.with_traceback(None)
2025-01-06T15:36:20.6482298Z E           numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6482879Z E           [1m[1m[1m[1mFailed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6483471Z E           [1m[1m[1mNo implementation of function Function(<built-in function concatenate>) found for signature:
2025-01-06T15:36:20.6484018Z E            
2025-01-06T15:36:20.6484288Z E            >>> concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))
2025-01-06T15:36:20.6484795Z E            
2025-01-06T15:36:20.6485012Z E           There are 2 candidate implementations:
2025-01-06T15:36:20.6485361Z E           [1m  - Of which 1 did not match due to:
2025-01-06T15:36:20.6485763Z E             Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:36:20.6486316Z E               With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)))':[0m
2025-01-06T15:36:20.6486815Z E           [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6487240Z E                TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6487839Z E              [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:36:20.6488248Z E               
2025-01-06T15:36:20.6488568Z E               >>> np_concatenate(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:36:20.6488932Z E               
2025-01-06T15:36:20.6489147Z E              There are 2 candidate implementations:
2025-01-06T15:36:20.6489489Z E              [1m      - Of which 2 did not match due to:
2025-01-06T15:36:20.6489891Z E                    Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:36:20.6490455Z E                      With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:36:20.6490941Z E              [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6491611Z E                       TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:36:20.6492381Z E                raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:36:20.6492901Z E              [0m
2025-01-06T15:36:20.6493289Z E              [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:36:20.6494028Z E              [0m[1mDuring: typing of call at /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:36:20.6494812Z E              [0m
2025-01-06T15:36:20.6495022Z E              [1m
2025-01-06T15:36:20.6495495Z E              File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:36:20.6496000Z E              [1m        def impl(arrays, axis=0):
2025-01-06T15:36:20.6496345Z E              [1m            return np_concatenate(arrays, axis)
2025-01-06T15:36:20.6496664Z E              [0m            [1m^[0m[0m
2025-01-06T15:36:20.6496921Z E           [0m
2025-01-06T15:36:20.6497346Z E             raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:36:20.6497896Z E           [1m  - Of which 1 did not match due to:
2025-01-06T15:36:20.6498290Z E             Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:36:20.6498830Z E               With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))':[0m
2025-01-06T15:36:20.6499297Z E           [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6499726Z E                TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6500334Z E              [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:36:20.6500743Z E               
2025-01-06T15:36:20.6501050Z E               >>> np_concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:36:20.6501404Z E               
2025-01-06T15:36:20.6501617Z E              There are 2 candidate implementations:
2025-01-06T15:36:20.6502089Z E              [1m      - Of which 1 did not match due to:
2025-01-06T15:36:20.6502484Z E                    Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:36:20.6503052Z E                      With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:36:20.6503543Z E              [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6504219Z E                       TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:36:20.6505176Z E                raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:36:20.6505751Z E              [1m      - Of which 1 did not match due to:
2025-01-06T15:36:20.6506141Z E                    Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:36:20.6506737Z E                      With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))':[0m
2025-01-06T15:36:20.6507243Z E              [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6507907Z E                       TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=[0]>, array(float64, 1d, C))[0m
2025-01-06T15:36:20.6508653Z E                raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:36:20.6509160Z E              [0m
2025-01-06T15:36:20.6509538Z E              [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:36:20.6510273Z E              [0m[1mDuring: typing of call at /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:36:20.6510807Z E              [0m
2025-01-06T15:36:20.6511014Z E              [1m
2025-01-06T15:36:20.6511482Z E              File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:36:20.6511987Z E              [1m        def impl(arrays, axis=0):
2025-01-06T15:36:20.6512331Z E              [1m            return np_concatenate(arrays, axis)
2025-01-06T15:36:20.6512791Z E              [0m            [1m^[0m[0m
2025-01-06T15:36:20.6513049Z E           [0m
2025-01-06T15:36:20.6513472Z E             raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:36:20.6513962Z E           [0m
2025-01-06T15:36:20.6514351Z E           [0m[1mDuring: resolving callee type: Function(<built-in function concatenate>)[0m
2025-01-06T15:36:20.6515240Z E           [0m[1mDuring: typing of call at /home/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (36)
2025-01-06T15:36:20.6515822Z E           [0m
2025-01-06T15:36:20.6516018Z E           [1m
2025-01-06T15:36:20.6516423Z E           File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 36:[0m
2025-01-06T15:36:20.6516899Z E           [1mdef numba_cumulative_trapezoid(f, x):
2025-01-06T15:36:20.6517173Z E               <source elided>
2025-01-06T15:36:20.6517392Z E           
2025-01-06T15:36:20.6517706Z E           [1m    return np.concatenate(([0], cumulative / cumulative[-1]))
2025-01-06T15:36:20.6518057Z E           [0m    [1m^[0m[0m
2025-01-06T15:36:20.6518275Z E           
2025-01-06T15:36:20.6518779Z E           [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x7f023c3bd080>))[0m
2025-01-06T15:36:20.6519670Z E           [0m[1mDuring: typing of call at /home/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:36:20.6520238Z E           [0m
2025-01-06T15:36:20.6520740Z E           [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x7f023c3bd080>))[0m
2025-01-06T15:36:20.6521746Z E           [0m[1mDuring: typing of call at /home/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:36:20.6522313Z E           [0m
2025-01-06T15:36:20.6522515Z E           [1m
2025-01-06T15:36:20.6522923Z E           File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 68:[0m
2025-01-06T15:36:20.6523480Z E           [1mdef cumulative_integrate_array_by_blocks(f, x, block_references):
2025-01-06T15:36:20.6523831Z E               <source elided>
2025-01-06T15:36:20.6524044Z E           
2025-01-06T15:36:20.6524367Z E           [1m            integrated[start:stop, col] = numba_cumulative_trapezoid(
2025-01-06T15:36:20.6524829Z E           [0m            [1m^[0m[0m
2025-01-06T15:36:20.6524993Z 
2025-01-06T15:36:20.6525275Z ../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/dispatcher.py:409: TypingError
2025-01-06T15:36:20.6525798Z ---------------------------- Captured stdout setup -----------------------------
2025-01-06T15:36:20.6526372Z [[1mtardis.io.model.parse_density_configuration[0m][[1;33mWARNING[0m]  
2025-01-06T15:36:20.6527217Z 	Number of density points larger than number of shells. Assuming inner point irrelevant ([1mparse_density_configuration.py[0m:114)
2025-01-06T15:36:20.6527909Z ------------------------------ Captured log setup ------------------------------
2025-01-06T15:36:20.6528952Z WARNING  tardis.io.model.parse_density_configuration:parse_density_configuration.py:114 Number of density points larger than number of shells. Assuming inner point irrelevant
2025-01-06T15:36:20.6529744Z =============================== warnings summary ===============================
2025-01-06T15:36:20.6530192Z tardis/io/configuration/config_validator.py:6
2025-01-06T15:36:20.6530625Z tardis/io/configuration/config_validator.py:6
2025-01-06T15:36:20.6531217Z   /home/runner/work/tardis/tardis/tardis/io/configuration/config_validator.py:6: DeprecationWarning:
2025-01-06T15:36:20.6531846Z   
2025-01-06T15:36:20.6533167Z   jsonschema.RefResolver is deprecated as of v4.18.0, in favor of the https://github.com/python-jsonschema/referencing library, which provides more compliant referencing behavior as well as more flexible APIs for customization. A future release will remove RefResolver. Please file a feature request (on referencing) if you are missing an API for the kind of customization you need.
2025-01-06T15:36:20.6534687Z 
2025-01-06T15:36:20.6546259Z ../../../micromamba/envs/tardis/lib/python3.12/site-packages/jupyter_client/connect.py:22
2025-01-06T15:36:20.6547065Z   /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/jupyter_client/connect.py:22: DeprecationWarning:
2025-01-06T15:36:20.6547591Z   
2025-01-06T15:36:20.6547842Z   Jupyter is migrating its paths to use standard platformdirs
2025-01-06T15:36:20.6548249Z   given by the platformdirs library.  To remove this warning and
2025-01-06T15:36:20.6548663Z   see the appropriate new directories, set the environment variable
2025-01-06T15:36:20.6549058Z   `JUPYTER_PLATFORM_DIRS=1` and then run `jupyter --paths`.
2025-01-06T15:36:20.6549441Z   The use of platformdirs will be the default in `jupyter_core` v6
2025-01-06T15:36:20.6549685Z 
2025-01-06T15:36:20.6554058Z tardis/io/tests/test_HDFWriter.py::test_multi_index_write
2025-01-06T15:36:20.6554773Z tardis/visualization/widgets/tests/test_shell_info.py::TestHDFShellInfo::test_shells_data
2025-01-06T15:36:20.6555377Z   /home/runner/work/tardis/tardis/tardis/io/util.py:276: PerformanceWarning:
2025-01-06T15:36:20.6555808Z   
2025-01-06T15:36:20.6555968Z   
2025-01-06T15:36:20.6556270Z   your performance may suffer as PyTables will pickle object types that it cannot
2025-01-06T15:36:20.6556838Z   map directly to c-types [inferred_type->mixed,key->block0_values] [items->Index([0], dtype='int64')]
2025-01-06T15:36:20.6557184Z 
2025-01-06T15:36:20.6557390Z tardis/io/tests/test_model_reader.py::test_read_cmfgen_isotope_abundances
2025-01-06T15:36:20.6558071Z tardis/plasma/tests/test_tardis_model_density_config.py::test_electron_densities
2025-01-06T15:36:20.6558620Z tardis/plasma/tests/test_tardis_model_density_config.py::test_isotope_number_densities
2025-01-06T15:36:20.6559127Z tardis/plasma/tests/test_tardis_model_density_config.py::test_t_rad
2025-01-06T15:36:20.6559702Z   /home/runner/work/tardis/tardis/tardis/io/model/readers/cmfgen_deprecated.py:82: DeprecationWarning:
2025-01-06T15:36:20.6560166Z   
2025-01-06T15:36:20.6560371Z   The current CMFGEN model parser is deprecated
2025-01-06T15:36:20.6560570Z 
2025-01-06T15:36:20.6560683Z tardis/io/tests/test_model_reader.py: 1 warning
2025-01-06T15:36:20.6561044Z tardis/plasma/tests/test_tardis_model_density_config.py: 9 warnings
2025-01-06T15:36:20.6561601Z   /home/runner/work/tardis/tardis/tardis/io/model/readers/cmfgen_deprecated.py:40: DeprecationWarning:
2025-01-06T15:36:20.6562057Z   
2025-01-06T15:36:20.6562255Z   The current CMFGEN model parser is deprecated
2025-01-06T15:36:20.6562446Z 
2025-01-06T15:36:20.6562653Z tardis/model/tests/test_base.py::TestModelFromPaper1Config::test_abundances
2025-01-06T15:36:20.6563299Z   /home/runner/work/tardis/tardis/tardis/transport/montecarlo/montecarlo_main_loop.py:123: NumbaTypeSafetyWarning:
2025-01-06T15:36:20.6563817Z   
2025-01-06T15:36:20.6564200Z   [1m[1m[1munsafe cast from uint64 to int64. Precision may be lost.[0m[0m[0m
2025-01-06T15:36:20.6564656Z 
2025-01-06T15:36:20.6564775Z tardis/model/tests/test_base.py: 6 warnings
2025-01-06T15:36:20.6565073Z tardis/model/tests/test_csvy_model.py: 2 warnings
2025-01-06T15:36:20.6565441Z tardis/plasma/tests/test_tardis_model_density_config.py: 3 warnings
2025-01-06T15:36:20.6565964Z   /home/runner/work/tardis/tardis/tardis/model/geometry/radial1d.py:50: DeprecationWarning:
2025-01-06T15:36:20.6566387Z   
2025-01-06T15:36:20.6566583Z   v_inner_boundary < 0, assuming default value
2025-01-06T15:36:20.6566767Z 
2025-01-06T15:36:20.6566878Z tardis/model/tests/test_base.py: 6 warnings
2025-01-06T15:36:20.6567167Z tardis/model/tests/test_csvy_model.py: 2 warnings
2025-01-06T15:36:20.6567532Z tardis/plasma/tests/test_tardis_model_density_config.py: 3 warnings
2025-01-06T15:36:20.6568053Z   /home/runner/work/tardis/tardis/tardis/model/geometry/radial1d.py:61: DeprecationWarning:
2025-01-06T15:36:20.6568608Z   
2025-01-06T15:36:20.6568797Z   v_outer_boundary < 0, assuming default value
2025-01-06T15:36:20.6568991Z 
2025-01-06T15:36:20.6569461Z tardis/plasma/equilibrium/tests/test_level_populations.py::TestLevelPopulationSolver::test_solve[collisional_rate_solver0-radiative_transitions0]
2025-01-06T15:36:20.6570290Z   /home/runner/work/tardis/tardis/tardis/plasma/equilibrium/level_populations.py:65: FutureWarning:
2025-01-06T15:36:20.6570742Z   
2025-01-06T15:36:20.6571109Z   A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
2025-01-06T15:36:20.6571907Z   The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.
2025-01-06T15:36:20.6572521Z   
2025-01-06T15:36:20.6573129Z   For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.
2025-01-06T15:36:20.6573722Z 
2025-01-06T15:36:20.6573875Z tardis/plasma/tests/test_complete_plasmas.py: 20 warnings
2025-01-06T15:36:20.6574503Z   /home/runner/work/tardis/tardis/tardis/plasma/tests/test_complete_plasmas.py:197: UserWarning:
2025-01-06T15:36:20.6574946Z   
2025-01-06T15:36:20.6575120Z   Property "atomic_mass" not found
2025-01-06T15:36:20.6575288Z 
2025-01-06T15:36:20.6575715Z tardis/plasma/tests/test_complete_plasmas.py::TestPlasma::test_plasma_properties[radiative_rates_type:detailed-w_epsilon:1e-10-j_blues_norm_factor]
2025-01-06T15:36:20.6576598Z tardis/plasma/tests/test_complete_plasmas.py::TestPlasma::test_plasma_properties[radiative_rates_type:detailed-j_blues_norm_factor]
2025-01-06T15:36:20.6577453Z   /home/runner/work/tardis/tardis/tardis/plasma/tests/test_complete_plasmas.py:197: UserWarning:
2025-01-06T15:36:20.6577884Z   
2025-01-06T15:36:20.6578067Z   Property "j_blues_norm_factor" not found
2025-01-06T15:36:20.6578254Z 
2025-01-06T15:36:20.6578668Z tardis/plasma/tests/test_complete_plasmas.py::TestPlasma::test_plasma_properties[radiative_rates_type:detailed-w_epsilon:1e-10-j_blue_estimator]
2025-01-06T15:36:20.6579569Z tardis/plasma/tests/test_complete_plasmas.py::TestPlasma::test_plasma_properties[radiative_rates_type:detailed-j_blue_estimator]
2025-01-06T15:36:20.6580304Z   /home/runner/work/tardis/tardis/tardis/plasma/tests/test_complete_plasmas.py:197: UserWarning:
2025-01-06T15:36:20.6580732Z   
2025-01-06T15:36:20.6580908Z   Property "j_blue_estimator" not found
2025-01-06T15:36:20.6581078Z 
2025-01-06T15:36:20.6581466Z tardis/plasma/tests/test_complete_plasmas.py::TestPlasma::test_plasma_properties[radiative_rates_type:detailed-w_epsilon:1e-10-volume]
2025-01-06T15:36:20.6582271Z tardis/plasma/tests/test_complete_plasmas.py::TestPlasma::test_plasma_properties[radiative_rates_type:detailed-volume]
2025-01-06T15:36:20.6582965Z   /home/runner/work/tardis/tardis/tardis/plasma/tests/test_complete_plasmas.py:197: UserWarning:
2025-01-06T15:36:20.6583394Z   
2025-01-06T15:36:20.6583564Z   Property "volume" not found
2025-01-06T15:36:20.6583715Z 
2025-01-06T15:36:20.6584104Z tardis/plasma/tests/test_complete_plasmas.py::TestPlasma::test_plasma_properties[radiative_rates_type:detailed-w_epsilon:1e-10-r_inner]
2025-01-06T15:36:20.6585167Z tardis/plasma/tests/test_complete_plasmas.py::TestPlasma::test_plasma_properties[radiative_rates_type:detailed-r_inner]
2025-01-06T15:36:20.6585862Z   /home/runner/work/tardis/tardis/tardis/plasma/tests/test_complete_plasmas.py:197: UserWarning:
2025-01-06T15:36:20.6586286Z   
2025-01-06T15:36:20.6586453Z   Property "r_inner" not found
2025-01-06T15:36:20.6586611Z 
2025-01-06T15:36:20.6586864Z tardis/plasma/tests/test_nlte_solver.py::test_critical_case_dilution_factor_1_root
2025-01-06T15:36:20.6587598Z   /home/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py:68: NumbaPendingDeprecationWarning:
2025-01-06T15:36:20.6588317Z   
2025-01-06T15:36:20.6589515Z   [1m[1m[1m[1mCode using Numba extension API maybe depending on 'old_style' error-capturing, which is deprecated and will be replaced by 'new_style' in a future release. See details at https://numba.readthedocs.io/en/latest/reference/deprecation.html#deprecation-of-old-style-numba-captured-errors
2025-01-06T15:36:20.6590545Z   Exception origin:
2025-01-06T15:36:20.6591092Z     File "/home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py", line 526, in _sequence_of_arrays
2025-01-06T15:36:20.6591758Z       raise TypeError("%s(): expecting a non-empty tuple of arrays, "
2025-01-06T15:36:20.6592101Z   [0m[0m[0m[0m
2025-01-06T15:36:20.6592224Z 
2025-01-06T15:36:20.6592439Z tardis/plasma/tests/test_plasma_vboundary.py::test_plasma_vboundary[2900-3750]
2025-01-06T15:36:20.6592942Z tardis/plasma/tests/test_plasma_vboundary.py::test_plasma_vboundary[2900-3850]
2025-01-06T15:36:20.6593429Z tardis/plasma/tests/test_plasma_vboundary.py::test_plasma_vboundary[2900-3900]
2025-01-06T15:36:20.6593958Z   /home/runner/work/tardis/tardis/tardis/model/geometry/radial1d.py:71: UserWarning:
2025-01-06T15:36:20.6594351Z   
2025-01-06T15:36:20.6594737Z   Requesting inner boundary below inner shell. Extrapolating the inner cell
2025-01-06T15:36:20.6595017Z 
2025-01-06T15:36:20.6595228Z tardis/spectrum/tests/test_spectrum_solver.py::TestSpectrumSolver::test_solve
2025-01-06T15:36:20.6595717Z   /home/runner/work/tardis/tardis/tardis/spectrum/base.py:70: UserWarning:
2025-01-06T15:36:20.6596072Z   
2025-01-06T15:36:20.6596481Z   SpectrumSolver.spectrum_virtual_packets is zero. Please run the montecarlo simulation with no_of_virtual_packets > 0
2025-01-06T15:36:20.6597013Z 
2025-01-06T15:36:20.6597228Z tardis/spectrum/tests/test_spectrum_solver.py::TestSpectrumSolver::test_solve
2025-01-06T15:36:20.6597719Z   /home/runner/work/tardis/tardis/tardis/spectrum/base.py:111: UserWarning:
2025-01-06T15:36:20.6598066Z   
2025-01-06T15:36:20.6598505Z   SpectrumSolver.integrator: The FormalIntegrator is not yet available.Please run the montecarlo simulation at least once.
2025-01-06T15:36:20.6598934Z 
2025-01-06T15:36:20.6599149Z tardis/transport/montecarlo/tests/test_vpacket.py::test_trace_vpacket_volley
2025-01-06T15:36:20.6599554Z   <string>:3: NumbaPendingDeprecationWarning:
2025-01-06T15:36:20.6599806Z   
2025-01-06T15:36:20.6600930Z   [1m[1mCode using Numba extension API maybe depending on 'old_style' error-capturing, which is deprecated and will be replaced by 'new_style' in a future release. See details at https://numba.readthedocs.io/en/latest/reference/deprecation.html#deprecation-of-old-style-numba-captured-errors
2025-01-06T15:36:20.6601941Z   Exception origin:
2025-01-06T15:36:20.6602309Z     File "/home/runner/micromamba/envs/tardis/lib/python3.12/inspect.py", line 3180, in _bind
2025-01-06T15:36:20.6602734Z       raise TypeError(msg) from None
2025-01-06T15:36:20.6602987Z   [0m[0m
2025-01-06T15:36:20.6603090Z 
2025-01-06T15:36:20.6603419Z tardis/visualization/tools/tests/test_sdec_plot.py::TestSDECPlotter::test_mpl_image[plotter_generate_plot_mpl4]
2025-01-06T15:36:20.6604110Z   /home/runner/work/tardis/tardis/tardis/visualization/tools/sdec_plot.py:1225: RuntimeWarning:
2025-01-06T15:36:20.6604638Z   
2025-01-06T15:36:20.6605571Z   More than 20 figures have been opened. Figures created through the pyplot interface (`matplotlib.pyplot.figure`) are retained until explicitly closed and may consume too much memory. (To control this warning, see the rcParam `figure.max_open_warning`). Consider using `matplotlib.pyplot.close()`.
2025-01-06T15:36:20.6606490Z 
2025-01-06T15:36:20.6606845Z tardis/visualization/widgets/tests/test_custom_abundance.py::TestCustomAbundanceWidget::test_update_input_item_value
2025-01-06T15:36:20.6607624Z tardis/visualization/widgets/tests/test_custom_abundance.py::TestCustomAbundanceWidget::test_update_input_item_value
2025-01-06T15:36:20.6608392Z tardis/visualization/widgets/tests/test_custom_abundance.py::TestCustomAbundanceWidget::test_update_input_item_value
2025-01-06T15:36:20.6609274Z tardis/visualization/widgets/tests/test_custom_abundance.py::TestCustomAbundanceWidget::test_update_input_item_value
2025-01-06T15:36:20.6610043Z tardis/visualization/widgets/tests/test_custom_abundance.py::TestCustomAbundanceWidget::test_update_input_item_value
2025-01-06T15:36:20.6610806Z tardis/visualization/widgets/tests/test_custom_abundance.py::TestCustomAbundanceWidget::test_update_input_item_value
2025-01-06T15:36:20.6611576Z   /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/traitlets/traitlets.py:1385: DeprecationWarning:
2025-01-06T15:36:20.6612084Z   
2025-01-06T15:36:20.6612359Z   Passing unrecognized arguments to super(Checkbox).__init__(icon='lock').
2025-01-06T15:36:20.6612812Z   object.__init__() takes exactly one argument (the instance to initialize)
2025-01-06T15:36:20.6613304Z   This is deprecated in traitlets 4.2.This error will be raised in a future release of traitlets.
2025-01-06T15:36:20.6613626Z 
2025-01-06T15:36:20.6613975Z tardis/visualization/widgets/tests/test_custom_abundance.py::TestCustomAbundanceWidget::test_update_input_item_value
2025-01-06T15:36:20.6614839Z tardis/visualization/widgets/tests/test_custom_abundance.py::TestCustomAbundanceWidget::test_update_input_item_value
2025-01-06T15:36:20.6615611Z   /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/traitlets/traitlets.py:1385: DeprecationWarning:
2025-01-06T15:36:20.6616105Z   
2025-01-06T15:36:20.6616364Z   Passing unrecognized arguments to super(FloatText).__init__(min=0).
2025-01-06T15:36:20.6616792Z   object.__init__() takes exactly one argument (the instance to initialize)
2025-01-06T15:36:20.6617279Z   This is deprecated in traitlets 4.2.This error will be raised in a future release of traitlets.
2025-01-06T15:36:20.6617715Z 
2025-01-06T15:36:20.6618066Z tardis/visualization/widgets/tests/test_custom_abundance.py::TestCustomAbundanceWidget::test_update_input_item_value
2025-01-06T15:36:20.6618832Z   /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/traitlets/traitlets.py:1385: DeprecationWarning:
2025-01-06T15:36:20.6619335Z   
2025-01-06T15:36:20.6619597Z   Passing unrecognized arguments to super(HTML).__init__(indent=True).
2025-01-06T15:36:20.6620026Z   object.__init__() takes exactly one argument (the instance to initialize)
2025-01-06T15:36:20.6620513Z   This is deprecated in traitlets 4.2.This error will be raised in a future release of traitlets.
2025-01-06T15:36:20.6620822Z 
2025-01-06T15:36:20.6621251Z tardis/visualization/widgets/tests/test_line_info.py::TestLineInfoWidgetData::test_get_species_interactions[wavelength_range0-packet_out_nu]
2025-01-06T15:36:20.6622073Z   /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/qgridnext/grid.py:830: DeprecationWarning:
2025-01-06T15:36:20.6622553Z   
2025-01-06T15:36:20.6622907Z   QgridWidget._show_toolbar_changed is deprecated in traitlets 4.1: use @observe and @unobserve instead.
2025-01-06T15:36:20.6623272Z 
2025-01-06T15:36:20.6623528Z tardis/visualization/widgets/tests/test_shell_info.py::TestHDFShellInfo::test_shells_data
2025-01-06T15:36:20.6624078Z   /home/runner/work/tardis/tardis/tardis/io/util.py:289: PerformanceWarning:
2025-01-06T15:36:20.6624615Z   
2025-01-06T15:36:20.6624771Z   
2025-01-06T15:36:20.6625058Z   your performance may suffer as PyTables will pickle object types that it cannot
2025-01-06T15:36:20.6625594Z   map directly to c-types [inferred_type->mixed,key->block0_values] [items->Index([0], dtype='int64')]
2025-01-06T15:36:20.6625932Z 
2025-01-06T15:36:20.6626182Z tardis/visualization/widgets/tests/test_shell_info.py::TestHDFShellInfo::test_shells_data
2025-01-06T15:36:20.6626724Z   /home/runner/work/tardis/tardis/tardis/io/util.py:292: PerformanceWarning:
2025-01-06T15:36:20.6627081Z   
2025-01-06T15:36:20.6627227Z   
2025-01-06T15:36:20.6627507Z   your performance may suffer as PyTables will pickle object types that it cannot
2025-01-06T15:36:20.6627961Z   map directly to c-types [inferred_type->mixed,key->values] [items->None]
2025-01-06T15:36:20.6628366Z 
2025-01-06T15:36:20.6628624Z tardis/visualization/widgets/tests/test_shell_info.py::TestHDFShellInfo::test_shells_data
2025-01-06T15:36:20.6629199Z   /home/runner/work/tardis/tardis/tardis/spectrum/formal_integral.py:398: UserWarning:
2025-01-06T15:36:20.6629592Z   
2025-01-06T15:36:20.6629874Z   The number of interpolate_shells was not specified. The value was set to 80.
2025-01-06T15:36:20.6630152Z 
2025-01-06T15:36:20.6630338Z -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
2025-01-06T15:36:20.6630607Z 
2025-01-06T15:36:20.6630768Z ---------- coverage: platform linux, python 3.12.4-final-0 -----------
2025-01-06T15:36:20.6631136Z Coverage HTML written to dir coverage_html_report
2025-01-06T15:36:20.6631430Z Coverage XML written to file coverage.xml
2025-01-06T15:36:20.6631610Z 
2025-01-06T15:36:20.6631731Z =========================== short test summary info ============================
2025-01-06T15:36:20.6632464Z ERROR tardis/plasma/tests/test_nlte_solver.py::test_critical_case_dilution_factor_1_root - numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6633302Z [1m[1m[1m[1mFailed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6633875Z [1m[1m[1mNo implementation of function Function(<built-in function concatenate>) found for signature:
2025-01-06T15:36:20.6634285Z  
2025-01-06T15:36:20.6634637Z  >>> concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))
2025-01-06T15:36:20.6634942Z  
2025-01-06T15:36:20.6635120Z There are 2 candidate implementations:
2025-01-06T15:36:20.6635420Z [1m  - Of which 1 did not match due to:
2025-01-06T15:36:20.6635953Z   Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:36:20.6636489Z     With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)))':[0m
2025-01-06T15:36:20.6636949Z [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6637363Z      TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6637942Z    [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:36:20.6638338Z     
2025-01-06T15:36:20.6638628Z     >>> np_concatenate(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:36:20.6638981Z     
2025-01-06T15:36:20.6639162Z    There are 2 candidate implementations:
2025-01-06T15:36:20.6639479Z    [1m      - Of which 2 did not match due to:
2025-01-06T15:36:20.6639856Z          Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:36:20.6640397Z            With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:36:20.6640867Z    [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6641514Z             TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:36:20.6642270Z      raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:36:20.6642768Z    [0m
2025-01-06T15:36:20.6643110Z    [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:36:20.6643809Z    [0m[1mDuring: typing of call at /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:36:20.6644322Z    [0m
2025-01-06T15:36:20.6644679Z    [1m
2025-01-06T15:36:20.6645128Z    File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:36:20.6645613Z    [1m        def impl(arrays, axis=0):
2025-01-06T15:36:20.6645929Z    [1m            return np_concatenate(arrays, axis)
2025-01-06T15:36:20.6646223Z    [0m            [1m^[0m[0m
2025-01-06T15:36:20.6646448Z [0m
2025-01-06T15:36:20.6646842Z   raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:36:20.6647494Z [1m  - Of which 1 did not match due to:
2025-01-06T15:36:20.6647867Z   Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:36:20.6648389Z     With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))':[0m
2025-01-06T15:36:20.6648828Z [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6649227Z      TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6649797Z    [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:36:20.6650198Z     
2025-01-06T15:36:20.6650474Z     >>> np_concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:36:20.6650816Z     
2025-01-06T15:36:20.6650996Z    There are 2 candidate implementations:
2025-01-06T15:36:20.6651309Z    [1m      - Of which 1 did not match due to:
2025-01-06T15:36:20.6651692Z          Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:36:20.6652232Z            With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:36:20.6652715Z    [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6653361Z             TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:36:20.6654103Z      raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:36:20.6654758Z    [1m      - Of which 1 did not match due to:
2025-01-06T15:36:20.6655251Z          Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:36:20.6655811Z            With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))':[0m
2025-01-06T15:36:20.6656294Z    [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6656932Z             TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=[0]>, array(float64, 1d, C))[0m
2025-01-06T15:36:20.6657665Z      raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:36:20.6658161Z    [0m
2025-01-06T15:36:20.6658507Z    [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:36:20.6659207Z    [0m[1mDuring: typing of call at /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:36:20.6659724Z    [0m
2025-01-06T15:36:20.6659897Z    [1m
2025-01-06T15:36:20.6660331Z    File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:36:20.6660804Z    [1m        def impl(arrays, axis=0):
2025-01-06T15:36:20.6661114Z    [1m            return np_concatenate(arrays, axis)
2025-01-06T15:36:20.6661411Z    [0m            [1m^[0m[0m
2025-01-06T15:36:20.6661635Z [0m
2025-01-06T15:36:20.6662023Z   raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:36:20.6662492Z [0m
2025-01-06T15:36:20.6662841Z [0m[1mDuring: resolving callee type: Function(<built-in function concatenate>)[0m
2025-01-06T15:36:20.6663598Z [0m[1mDuring: typing of call at /home/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (36)
2025-01-06T15:36:20.6664155Z [0m
2025-01-06T15:36:20.6664319Z [1m
2025-01-06T15:36:20.6664875Z File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 36:[0m
2025-01-06T15:36:20.6665333Z [1mdef numba_cumulative_trapezoid(f, x):
2025-01-06T15:36:20.6665582Z     <source elided>
2025-01-06T15:36:20.6665698Z 
2025-01-06T15:36:20.6665901Z [1m    return np.concatenate(([0], cumulative / cumulative[-1]))
2025-01-06T15:36:20.6666349Z [0m    [1m^[0m[0m
2025-01-06T15:36:20.6666466Z 
2025-01-06T15:36:20.6666863Z [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x7f023c3bd080>))[0m
2025-01-06T15:36:20.6667731Z [0m[1mDuring: typing of call at /home/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:36:20.6668285Z [0m
2025-01-06T15:36:20.6668754Z [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x7f023c3bd080>))[0m
2025-01-06T15:36:20.6669606Z [0m[1mDuring: typing of call at /home/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:36:20.6670156Z [0m
2025-01-06T15:36:20.6670321Z [1m
2025-01-06T15:36:20.6670691Z File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 68:[0m
2025-01-06T15:36:20.6671225Z [1mdef cumulative_integrate_array_by_blocks(f, x, block_references):
2025-01-06T15:36:20.6671550Z     <source elided>
2025-01-06T15:36:20.6671671Z 
2025-01-06T15:36:20.6671872Z [1m            integrated[start:stop, col] = numba_cumulative_trapezoid(
2025-01-06T15:36:20.6672211Z [0m            [1m^[0m[0m
2025-01-06T15:36:20.6672866Z ERROR tardis/plasma/tests/test_nlte_solver.py::test_critical_case_dilution_factor_1_lu - numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6673674Z [1m[1m[1m[1mFailed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6674239Z [1m[1m[1mNo implementation of function Function(<built-in function concatenate>) found for signature:
2025-01-06T15:36:20.6674761Z  
2025-01-06T15:36:20.6675120Z  >>> concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))
2025-01-06T15:36:20.6675423Z  
2025-01-06T15:36:20.6675600Z There are 2 candidate implementations:
2025-01-06T15:36:20.6675898Z [1m  - Of which 1 did not match due to:
2025-01-06T15:36:20.6676273Z   Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:36:20.6676810Z     With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)))':[0m
2025-01-06T15:36:20.6677256Z [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6677660Z      TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6678235Z    [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:36:20.6678630Z     
2025-01-06T15:36:20.6678916Z     >>> np_concatenate(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:36:20.6679271Z     
2025-01-06T15:36:20.6679453Z    There are 2 candidate implementations:
2025-01-06T15:36:20.6679770Z    [1m      - Of which 2 did not match due to:
2025-01-06T15:36:20.6680144Z          Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:36:20.6680686Z            With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:36:20.6681168Z    [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6681817Z             TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:36:20.6682557Z      raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:36:20.6683048Z    [0m
2025-01-06T15:36:20.6683393Z    [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:36:20.6684095Z    [0m[1mDuring: typing of call at /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:36:20.6684721Z    [0m
2025-01-06T15:36:20.6684896Z    [1m
2025-01-06T15:36:20.6685334Z    File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:36:20.6685808Z    [1m        def impl(arrays, axis=0):
2025-01-06T15:36:20.6686242Z    [1m            return np_concatenate(arrays, axis)
2025-01-06T15:36:20.6686535Z    [0m            [1m^[0m[0m
2025-01-06T15:36:20.6686787Z [0m
2025-01-06T15:36:20.6687180Z   raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:36:20.6687695Z [1m  - Of which 1 did not match due to:
2025-01-06T15:36:20.6688065Z   Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:36:20.6688586Z     With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))':[0m
2025-01-06T15:36:20.6689023Z [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6689435Z      TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6690021Z    [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:36:20.6690424Z     
2025-01-06T15:36:20.6690703Z     >>> np_concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:36:20.6691054Z     
2025-01-06T15:36:20.6691235Z    There are 2 candidate implementations:
2025-01-06T15:36:20.6691548Z    [1m      - Of which 1 did not match due to:
2025-01-06T15:36:20.6691919Z          Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:36:20.6692452Z            With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:36:20.6692914Z    [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6693557Z             TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:36:20.6694527Z      raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:36:20.6695080Z    [1m      - Of which 1 did not match due to:
2025-01-06T15:36:20.6695453Z          Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:36:20.6696017Z            With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))':[0m
2025-01-06T15:36:20.6696496Z    [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6697130Z             TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=[0]>, array(float64, 1d, C))[0m
2025-01-06T15:36:20.6697871Z      raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:36:20.6698359Z    [0m
2025-01-06T15:36:20.6698695Z    [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:36:20.6699395Z    [0m[1mDuring: typing of call at /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:36:20.6699908Z    [0m
2025-01-06T15:36:20.6700083Z    [1m
2025-01-06T15:36:20.6700522Z    File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:36:20.6700995Z    [1m        def impl(arrays, axis=0):
2025-01-06T15:36:20.6701307Z    [1m            return np_concatenate(arrays, axis)
2025-01-06T15:36:20.6701599Z    [0m            [1m^[0m[0m
2025-01-06T15:36:20.6701823Z [0m
2025-01-06T15:36:20.6702214Z   raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:36:20.6702696Z [0m
2025-01-06T15:36:20.6703045Z [0m[1mDuring: resolving callee type: Function(<built-in function concatenate>)[0m
2025-01-06T15:36:20.6703802Z [0m[1mDuring: typing of call at /home/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (36)
2025-01-06T15:36:20.6704357Z [0m
2025-01-06T15:36:20.6704776Z [1m
2025-01-06T15:36:20.6705161Z File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 36:[0m
2025-01-06T15:36:20.6705759Z [1mdef numba_cumulative_trapezoid(f, x):
2025-01-06T15:36:20.6706009Z     <source elided>
2025-01-06T15:36:20.6706125Z 
2025-01-06T15:36:20.6706324Z [1m    return np.concatenate(([0], cumulative / cumulative[-1]))
2025-01-06T15:36:20.6706648Z [0m    [1m^[0m[0m
2025-01-06T15:36:20.6706764Z 
2025-01-06T15:36:20.6707156Z [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x7f023c3bd080>))[0m
2025-01-06T15:36:20.6708020Z [0m[1mDuring: typing of call at /home/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:36:20.6708567Z [0m
2025-01-06T15:36:20.6709027Z [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x7f023c3bd080>))[0m
2025-01-06T15:36:20.6709889Z [0m[1mDuring: typing of call at /home/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:36:20.6710435Z [0m
2025-01-06T15:36:20.6710607Z [1m
2025-01-06T15:36:20.6711010Z File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 68:[0m
2025-01-06T15:36:20.6711541Z [1mdef cumulative_integrate_array_by_blocks(f, x, block_references):
2025-01-06T15:36:20.6711867Z     <source elided>
2025-01-06T15:36:20.6711985Z 
2025-01-06T15:36:20.6712186Z [1m            integrated[start:stop, col] = numba_cumulative_trapezoid(
2025-01-06T15:36:20.6712525Z [0m            [1m^[0m[0m
2025-01-06T15:36:20.6713190Z ERROR tardis/plasma/tests/test_nlte_solver.py::test_critical_case_dilution_factor_0_root - numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6714125Z [1m[1m[1m[1mFailed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6714804Z [1m[1m[1mNo implementation of function Function(<built-in function concatenate>) found for signature:
2025-01-06T15:36:20.6715218Z  
2025-01-06T15:36:20.6715458Z  >>> concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))
2025-01-06T15:36:20.6715766Z  
2025-01-06T15:36:20.6715949Z There are 2 candidate implementations:
2025-01-06T15:36:20.6716245Z [1m  - Of which 1 did not match due to:
2025-01-06T15:36:20.6716619Z   Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:36:20.6717146Z     With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)))':[0m
2025-01-06T15:36:20.6717594Z [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6718004Z      TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6718580Z    [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:36:20.6718979Z     
2025-01-06T15:36:20.6719266Z     >>> np_concatenate(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:36:20.6719621Z     
2025-01-06T15:36:20.6719807Z    There are 2 candidate implementations:
2025-01-06T15:36:20.6720124Z    [1m      - Of which 2 did not match due to:
2025-01-06T15:36:20.6720496Z          Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:36:20.6721034Z            With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:36:20.6721510Z    [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6722150Z             TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:36:20.6722893Z      raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:36:20.6723390Z    [0m
2025-01-06T15:36:20.6723730Z    [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:36:20.6724598Z    [0m[1mDuring: typing of call at /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:36:20.6725295Z    [0m
2025-01-06T15:36:20.6725475Z    [1m
2025-01-06T15:36:20.6725921Z    File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:36:20.6726403Z    [1m        def impl(arrays, axis=0):
2025-01-06T15:36:20.6726719Z    [1m            return np_concatenate(arrays, axis)
2025-01-06T15:36:20.6727013Z    [0m            [1m^[0m[0m
2025-01-06T15:36:20.6727235Z [0m
2025-01-06T15:36:20.6727633Z   raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:36:20.6728149Z [1m  - Of which 1 did not match due to:
2025-01-06T15:36:20.6728517Z   Overload in function 'impl_np_concatenate': File: numba/np/arrayobj.py: Line 5873.
2025-01-06T15:36:20.6729086Z     With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)))':[0m
2025-01-06T15:36:20.6729532Z [1m   Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6729951Z      TypingError: Failed in nopython mode pipeline (step: nopython frontend)
2025-01-06T15:36:20.6730534Z    [1m[1m[1mNo implementation of function Function(<intrinsic np_concatenate>) found for signature:
2025-01-06T15:36:20.6730934Z     
2025-01-06T15:36:20.6731213Z     >>> np_concatenate(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))
2025-01-06T15:36:20.6731561Z     
2025-01-06T15:36:20.6731746Z    There are 2 candidate implementations:
2025-01-06T15:36:20.6732062Z    [1m      - Of which 1 did not match due to:
2025-01-06T15:36:20.6732436Z          Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:36:20.6732977Z            With argument(s): '(Tuple(list(int64)<iv=None>, array(float64, 1d, C)), int64)':[0m
2025-01-06T15:36:20.6733568Z    [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6734219Z             TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=None>, array(float64, 1d, C))[0m
2025-01-06T15:36:20.6735082Z      raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:36:20.6735629Z    [1m      - Of which 1 did not match due to:
2025-01-06T15:36:20.6735998Z          Intrinsic in function 'np_concatenate': File: numba/np/arrayobj.py: Line 5856.
2025-01-06T15:36:20.6736560Z            With argument(s): '(Tuple(list(int64)<iv=[0]>, array(float64, 1d, C)), Literal[int](0))':[0m
2025-01-06T15:36:20.6737040Z    [1m       Rejected as the implementation raised a specific error:
2025-01-06T15:36:20.6737680Z             TypeError: np.concatenate(): expecting a non-empty tuple of arrays, got Tuple(list(int64)<iv=[0]>, array(float64, 1d, C))[0m
2025-01-06T15:36:20.6738420Z      raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typing/npydecl.py:526
2025-01-06T15:36:20.6738912Z    [0m
2025-01-06T15:36:20.6739255Z    [0m[1mDuring: resolving callee type: Function(<intrinsic np_concatenate>)[0m
2025-01-06T15:36:20.6739958Z    [0m[1mDuring: typing of call at /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py (5877)
2025-01-06T15:36:20.6740469Z    [0m
2025-01-06T15:36:20.6740642Z    [1m
2025-01-06T15:36:20.6741076Z    File "../../../micromamba/envs/tardis/lib/python3.12/site-packages/numba/np/arrayobj.py", line 5877:[0m
2025-01-06T15:36:20.6741549Z    [1m        def impl(arrays, axis=0):
2025-01-06T15:36:20.6741860Z    [1m            return np_concatenate(arrays, axis)
2025-01-06T15:36:20.6742154Z    [0m            [1m^[0m[0m
2025-01-06T15:36:20.6742379Z [0m
2025-01-06T15:36:20.6742776Z   raised from /home/runner/micromamba/envs/tardis/lib/python3.12/site-packages/numba/core/typeinfer.py:1091
2025-01-06T15:36:20.6743254Z [0m
2025-01-06T15:36:20.6743604Z [0m[1mDuring: resolving callee type: Function(<built-in function concatenate>)[0m
2025-01-06T15:36:20.6744364Z [0m[1mDuring: typing of call at /home/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (36)
2025-01-06T15:36:20.6745256Z [0m
2025-01-06T15:36:20.6745436Z [1m
2025-01-06T15:36:20.6745821Z File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 36:[0m
2025-01-06T15:36:20.6746268Z [1mdef numba_cumulative_trapezoid(f, x):
2025-01-06T15:36:20.6746520Z     <source elided>
2025-01-06T15:36:20.6746640Z 
2025-01-06T15:36:20.6746841Z [1m    return np.concatenate(([0], cumulative / cumulative[-1]))
2025-01-06T15:36:20.6747166Z [0m    [1m^[0m[0m
2025-01-06T15:36:20.6747283Z 
2025-01-06T15:36:20.6747681Z [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x7f023c3bd080>))[0m
2025-01-06T15:36:20.6748551Z [0m[1mDuring: typing of call at /home/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:36:20.6749099Z [0m
2025-01-06T15:36:20.6749572Z [0m[1mDuring: resolving callee type: type(CPUDispatcher(<function numba_cumulative_trapezoid at 0x7f023c3bd080>))[0m
2025-01-06T15:36:20.6750436Z [0m[1mDuring: typing of call at /home/runner/work/tardis/tardis/tardis/plasma/properties/continuum_processes/fast_array_util.py (68)
2025-01-06T15:36:20.6750980Z [0m
2025-01-06T15:36:20.6751145Z [1m
2025-01-06T15:36:20.6751512Z File "tardis/plasma/properties/continuum_processes/fast_array_util.py", line 68:[0m
2025-01-06T15:36:20.6752042Z [1mdef cumulative_integrate_array_by_blocks(f, x, block_references):
2025-01-06T15:36:20.6752370Z     <source elided>
2025-01-06T15:36:20.6752488Z 
2025-01-06T15:36:20.6752689Z [1m            integrated[start:stop, col] = numba_cumulative_trapezoid(
2025-01-06T15:36:20.6753035Z [0m            [1m^[0m[0m
2025-01-06T15:36:20.6753555Z = 1435 passed, 329 skipped, 2 deselected, 20 xfailed, 7 xpassed, 92 warnings, 3 errors in 620.16s (0:10:20) =
2025-01-06T15:36:22.7646865Z ##[error]Process completed with exit code 1.
